client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx

```diff
@@ -184,7 +188,9 @@ const VerticalSelect: React.FunctionComponent< Props > = ( { onNext } ) => {
 
 	React.useEffect( () => {
 		inputRef.current.innerText = siteVertical?.label || '';
-	}, [ siteVertical, inputRef ] );
+		const isFreeForm = ! siteVertical?.slug && !! siteVertical?.label.length;
+		recordVerticalSelection( getSelectedVerticalUntranslatedLabel(), isFreeForm );
+	}, [ siteVertical, inputRef ] ); //eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Shouldn't we pass `getSelectedVerticalUntranslatedLabel` as a dep too?
```suggestion
	}, [ siteVertical, inputRef, getSelectedVerticalUntranslatedLabel ] );
```